pkg/collector/runner/runner_test.go

```diff
@@ -23,6 +24,14 @@ type TestCheck struct {
 	sync.Mutex
```

Comment:
Is the mutex still useful with the channel?

Reply:
We still use `hasRun` in other tests (ie num_worker_test) so I think so

Reply:
Yes right